fhir-database-utils/src/main/java/com/ibm/fhir/database/utils/model/ProcedureDef.java

```diff
@@ -44,17 +40,23 @@ public void apply(IDatabaseAdapter target) {
         }
     }
 
-    /* (non-Javadoc)
-     * @see com.ibm.fhir.database.utils.model.IDatabaseObject#drop(com.ibm.fhir.database.utils.api.IDatabaseAdapter)
-     */
+    @Override
+    public void apply(Integer priorVersion, IDatabaseAdapter target) {
+        // Serialize the execution of the procedure, to try and avoid the
```

Comment:
how does synchronized help with deadlocks here? it's synchronized in java code? I realize you are not the original person to author it... but if you know...

Reply:
i guess the idea is just to allow a single thread to call `createOrReplaceProcedure` at a time (for a given target at least)...i wonder if we should do something similar for adding foreign key constraints...

Reply:
that's a good point... now we thought it out, let's monitor it and make it consistent if we need to